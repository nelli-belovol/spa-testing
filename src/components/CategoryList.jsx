import { CategoryItem } from './CategoryItem';

function CategoryList({ catalog = [] }) {
  return (
    <div className="list" role="list">
      {catalog.map((el) => (
        <CategoryItem key={el.idCategory} {...el} />
      ))}
    </div>
  );
}

export { CategoryList };
