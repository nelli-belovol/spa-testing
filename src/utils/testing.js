import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom"; // MemoryRouter was special created for testing 


export const renderWithRouter=(component, options={})=>{
	const Wrapper = ({children})=>(<MemoryRouter {...options}>{children}</MemoryRouter>)

	render(component, {wrapper: Wrapper})
}