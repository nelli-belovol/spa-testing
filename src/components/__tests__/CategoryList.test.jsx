import { screen } from '@testing-library/react';

import { CategoryList } from '../CategoryList';
import { renderWithRouter } from '../../utils/testing';

describe('CategoryItem', () => {
  it('should render correctly', () => {
    renderWithRouter(
      <CategoryList
        catalog={[
          {
            strCategory: 'Beef',
            strCategoryThumb:
              'https://www.themealdb.com/images/category/beef.png',
            strCategoryDescription:
              'Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times. Beef is a source of protein and nutrients.',
          },
          {
            strCategory: 'Chicken',
            strCategoryThumb:
              'https://www.themealdb.com/images/category/chicken.png',
            strCategoryDescription:
              'Chicken is the most common type of poultry in the world',
          },
        ]}
      />,
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
  it('shpuld render correctly with no items', () => {
    renderWithRouter(<CategoryList />);
    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});
