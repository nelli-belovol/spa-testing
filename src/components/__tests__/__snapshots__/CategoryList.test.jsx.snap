// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem should render correctly 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Beef"
        src="https://www.themealdb.com/images/category/beef.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Beef
      </span>
      <p>
        Beef is the culinary name for meat from cattle, particularly
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/Beef"
      >
        Watch category
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Chicken"
        src="https://www.themealdb.com/images/category/chicken.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Chicken
      </span>
      <p>
        Chicken is the most common type of poultry in the world
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/Chicken"
      >
        Watch category
      </a>
    </div>
  </div>
</div>
`;

exports[`CategoryItem shpuld render correctly with no items 1`] = `
<div
  class="list"
  role="list"
/>
`;
